src/sqlfluff/core/parser/lexer.py

```diff
@@ -74,12 +74,10 @@ def _trim(self, matched, start_pos):
         idx = 0
 
         if self.trim_post_subdivide:
-            trimmer = re.compile(self.trim_post_subdivide["regex"], re.DOTALL)
-            TrimClass = RawSegment.make(
-                self.trim_post_subdivide["regex"],
-                name=self.trim_post_subdivide["name"],
-                type=self.trim_post_subdivide["type"],
-            )
+            class_kwargs = self.trim_post_subdivide.copy()
```

Comment:
What are these changes? To my untrained eye, the changes are too subtle to grasp. 😄

Reply:
we extract the `regex` element out of the dict, and then use the rest as kwargs :)